python/cudf/cudf/core/udf/groupby_function.py

```diff
@@ -0,0 +1,867 @@
+# Copyright (c) 2020-2022, NVIDIA CORPORATION.
+
+import glob
+import math
+import os
+
+import cupy as cp
+import numba
+import numpy as np
+from numba import cuda, types
+from numba.core import cgutils
+from numba.core.extending import (
+    lower_builtin,
+    make_attribute_wrapper,
+    models,
+    register_model,
+    type_callable,
+    typeof_impl,
+)
+from numba.core.typing import signature as nb_signature
+from numba.core.typing.templates import AbstractTemplate, AttributeTemplate
+from numba.cuda.cudadecl import registry as cuda_registry
+from numba.cuda.cudadrv.devices import get_context
+from numba.cuda.cudaimpl import lower as cuda_lower
+from numba.np import numpy_support
+from numba.types import Record
+
+from cudf.core.column import as_column
+from cudf.core.udf.templates import (
+    group_initializer_template,
+    groupby_apply_kernel_template,
+)
+from cudf.core.udf.utils import (
+    _all_dtypes_from_frame,
+    _compile_or_get,
+    _get_kernel_groupby_apply,
+    _get_udf_return_type,
+    _supported_cols_from_frame,
+    _supported_dtypes_from_frame,
```

Comment:
This function denotes dtypes supported by `DataFrame.apply` and `Series.apply`, which isn't necessarily the same set of dtypes that are supported by `Groupby.apply`. Can we create the supported dtypes as a set and keep them as a global variable, then somehow generalize these functions to accept a valid set as argument? This is important especially as we don't support too many dtypes yet.